### PR TITLE
Split dev-tools/globals* into sim and server

### PR DESCRIPTION
**Pull request by scheibo**
All this PR accomplishes is moving things out of `dev-tools/` and into more 'logical' locations (which will have the side effect of letting us eventually remove the `dev-tools` directory). It is important to note that this is very much '[rearranging the deck chairs on the Titanic](https://en.wiktionary.org/wiki/rearrange_the_deck_chairs_on_the_Titanic)': **`sim`/`data` (and `lib`) still implicitly depend on some declarations in `server/globals*`**, `tsc` treats this change exactly as they were all in the same file anyway. Some concrete examples: code in `data/` still depends on `Monitor` declared in `server/globals.ts`, `lib/process-manager.ts` needs `ChildProcess` declared in `server/globals.d.ts` etc.

This change does almost nothing to help us properly modularize `sim` (design doc forthcoming) or eliminate global usage in our application (also a very worthwhile goal that will be pursued in the near future). **This is purely meant to be an incremental improvement on our current state.**

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -68,6 +68,6 @@
     "rulesDirectory": [],
     "linterOptions": {
       // TODO: why is tslint looking at .js files anyway?
-      "exclude": ["node_modules/", "**/*.js", "dev-tools/global*"]
+      "exclude": ["node_modules/", "**/*.js", "**/globals*.ts"]
```
Comment (Zarel) [inline tslint.json]:
```suggestion
      "exclude": ["node_modules/", "**/*.js", "*/globals.ts"]
```

seems equivalent?

Comment (scheibo) [reply tslint.json]: `globals.d.ts`?

Comment (scheibo) [reply tslint.json]: Can probably remove the `**` though

Comment (Zarel) [reply tslint.json]: I'm confused that `globals*.ts` would match `global.d.ts` – the latter is missing the first `s`.

Comment (Zarel) [reply tslint.json]: Also: is `**/*.js` necessary? Is `*.js` enough?

Comment (scheibo) [reply tslint.json]: `**/*.js` is necessary, and you're right, I had meant `global*.ts` and typoed but clearly it didn't matter. Maybe it skips `.d.ts` files by default (and yet I can't get it to exclude `.js` files? the irony...)?

Updated to be `*/globals.ts`

**Comment by Zarel** [comment]
Man, your merges do horrible things to the "Files changed" panel. :(

**Pull request merged**